### PR TITLE
Fix delay in log retrieval

### PR DESCRIPTION
**Pull request by hjmodi**
* CLOUDBLD-10225

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

File: osbs/tekton.py [hunk 1]
```diff
@@ -683,25 +683,31 @@ def wait_for_taskruns(self):
...
         """
-        watched_task_runs = []
+        watched_task_runs = set()
         for pipeline_run in self.os.watch_resource(
                 self.api_path,
                 self.api_version,
                 resource_type="pipelineruns",
                 resource_name=self.pipeline_run_name,
         ):
+            try:
```
Comment (rcerven) [inline osbs/tekton.py:693]: just use https://github.com/containerbuildsystem/osbs-client/blob/master/osbs/tekton.py#L553

File: osbs/tekton.py [hunk 2]
```diff
@@ -683,25 +683,31 @@ def wait_for_taskruns(self):
...
             except KeyError:
                 logger.error("pipeline run does not have any task runs")
                 continue
             for task_run_name, task_run_data in task_runs.items():
                 if task_run_name not in watched_task_runs:
-                    watched_task_runs.append(task_run_name)
+                    watched_task_runs.add(task_run_name)
                     yield task_run_data['pipelineTaskName'], task_run_name
-            # all task runs accounted for
-            if len(pipeline_run['status']['pipelineSpec']['tasks']) == len(task_runs):
```
[the remark this answers is not in the document]
Comment (hjmodi) [reply osbs/tekton.py:703]: moved

**Review by rcerven — APPROVED** [review]

File: osbs/tekton.py [hunk 3]
```diff
@@ -699,6 +699,14 @@ def wait_for_taskruns(self):
                 if task_run_name not in watched_task_runs:
                     watched_task_runs.add(task_run_name)
                     yield task_run_data['pipelineTaskName'], task_run_name
+            try:
+                status = pipeline_run['status']['conditions'][0]['status']
+            except KeyError:
+                logger.error("pipeline run does not have any status")
+                return
```
Comment (chmeliik) [inline osbs/tekton.py:706]: Should we `return` here? The other method (`wait_for_start`) keeps waiting if there's no status

File: tests/test_tekton.py [hunk 4]
```diff
@@ -820,20 +821,25 @@ def test_wait_for_start(self, pipeline_run):
         responses.add(responses.GET, PIPELINE_RUN_URL, json=PIPELINE_RUN_JSON)
         resp = pipeline_run.wait_for_start()
 
-        assert len(responses.calls) == 2
```
Comment (chmeliik) [inline tests/test_tekton.py:823]: Why the changes in this test? There were no changes in the method

(I'm guessing the `time.sleep()` mocking is to make things faster, but what about this one?)

File: osbs/tekton.py [hunk 5]
```diff
@@ -697,11 +697,8 @@ def wait_for_taskruns(self):
                 continue
             for task_run_name, task_run_data in task_runs.items():
                 if task_run_name not in watched_task_runs:
-                    watched_task_runs.append(task_run_name)
+                    watched_task_runs.add(task_run_name)
                     yield task_run_data['pipelineTaskName'], task_run_name
-            # all task runs accounted for
-            if len(pipeline_run['status']['pipelineSpec']['tasks']) == len(task_runs):
-                return
```
Comment (chmeliik) [inline osbs/tekton.py:704]: Now that I think about it, this commit on its own only makes the problem worse, by removing the only early-return condition that this method had. It's fixed in the next one, but maybe it should be a single commit.

File: osbs/tekton.py (hunk 3 above)
Comment (rcerven) [reply osbs/tekton.py:707]: wait_for_start is waiting for prun to start, and AT this point in wait_for_tasks, prun already started so it has status, and if not something broke with prun I'd say

Comment (chmeliik) [reply osbs/tekton.py:707]: Hmm, maybe raise an error then? Or don't even catch the KeyError and let it fail?

Comment (chmeliik) [reply osbs/tekton.py:707]: Hmm, alright then. I don't like ignoring this kind of error (one that should never happen), but fair enough :)

**Review by chmeliik — APPROVED** [review]
LGTM :+1:
